documentation/projects/proposals/project_improvement/20230914-implementation_plan_project_automations.md

```diff
@@ -0,0 +1,262 @@
+   the `state_reason` field inside the issue payload of the `closed` activity of
+   the
+   [`issues` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issues).
+
+3. [Issues/Issue is closed and inactive](/meta/project_boards/issues.md#issue-is-closed-and-inactive)
+
+   The current system means that issues are archived one by one as they become
+   inactive and with an 8 day cadence, this will slowly drift out of sync with
+   the weekly development chat.
+
+   Archiving should be performed by a custom scheduled workflow that archives
+   all items in the "✅ Done" and "🗑️ Discarded" columns weekly (preferably
+   immediately after the chat).
+
+4. [PRs/PR is created](/meta/project_boards/prs.md#pr-is-created) and
+   [PRs/PR is added to project](/meta/project_boards/prs.md#pr-is-added-to-the-project)
+
+   The PR should be added to the project board under the appropriate column, "In
+   Progress" or "Needs &lt;x&gt; Review(s)" depending on the PR's draft or
+   ready-for-review state and the number of existing reviews.
+
+5. [PRs/PR is closed or merged](/meta/project_boards/prs.md#pr-is-closed-or-merged)
+
+   The PR should be moved to the "✅ Done" column only if it was merged. If it
+   was closed without merge, it should be moved to a new "🗑️ Discarded" column.
+
+   This can be implemented by updating built-in workflows.
+
+### Restoration of previous automations
+
+1. Issues/Issue is connected to PR
+
+   When an issue is
+   [connected](https://docs.github.com/en/rest/overview/issue-event-types?apiVersion=2022-11-28#connected)
+   to a PR, the issue should move from "📋 Backlog" or "📅 To do" to the "🏗 In
+   Progress" column.
+
+   Unfortunately, the `connected` event does not trigger workflows. So this can
+   be implemented using a combination of multiple workflows.
+
+   - scheduled workflow
+   - [`pull_request` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request)
+     - created
+     - edited
+
+2. Issues/PR linked to issue is closed without merge
+
+   When the PR for an issue in "🏗️ In Progress" is closed without merge, the
+   issue should go back to the "📅 To do" column.
+
+   The `closed` activity in the
+   [`pull_request` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request)
+   can trigger workflows to do this.
+
+### Creation of new automations
+
+1. Issues/Issue is relabeled
+
+   When an issue's labels change, it should automatically move to the right
+   column.
+
+   - issues labeled as "⛔ status: blocked" ⟹ Move into "⛔ Blocked" column
+   - issues unlabeled as "⛔ status: blocked" ⟹ Move into "📋 Backlog" column
+
+   Both `labeled` and `unlabeled` activities in the
+   [`issues` event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issues)
+   can trigger workflows to do this.
+
+   The priority field in the project board issues can also be synced to the
```

Comment:
```suggestion
   The priority field in the project board issues should also be synced to the
```
?

Reply:
Changed in b8f90c555e8b76e7ef7c9cd6ec0a01c0d2cf7417.